File: jarvis/test-mcp.mjs
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";

async function run() {
  const transport = new StdioClientTransport({
    command: "C:\\Python313\\python.exe",
    args: ["e:\\cloning_project\\face-identification\\chrome-mcp\\src\\chrome_mcp\\server.py"]
  });

  const client = new Client({
    name: "jarvis-mcp-client",
    version: "1.0.0"
  }, {
    capabilities: {}
  });

  await client.connect(transport);
  console.log("Connected to MCP server!");
  
  const tools = await client.listTools();
  console.log("Available tools:", tools.tools.map(t => t.name));

  await client.close();
}

run().catch(console.error);

File: chrome-mcp/test_mcp.py
import asyncio
import os
import sys

async def main():
    # Use mcp client to connect to the local server
    from mcp.client.stdio import stdio_client, StdioServerParameters
    from mcp.client.session import ClientSession

    server_params = StdioServerParameters(
        command="python",
        args=["-m", "chrome_mcp.server"]
    )

    print("Connecting to MCP server...")
    async with stdio_client(server_params) as (read_stream, write_stream):
        async with ClientSession(read_stream, write_stream) as session:
            await session.initialize()
            print("Connected and initialized.")
            
            # List tools
            tools = await session.list_tools()
            print(f"Available tools: {[t.name for t in tools.tools]}")
            
            # Navigate to google
            print("Navigating to google.com...")
            result = await session.call_tool("navigate", {"url": "https://www.google.com"})
            print(f"Result: {result.content[0].text}")
            
            # Wait for page to load
            await asyncio.sleep(2)
            
            # Get title
            print("Getting title...")
            title_result = await session.call_tool("get_title", {})
            print(f"Title: {title_result.content[0].text}")
            
            # Click something maybe? No need, title is enough for a basic test.
            
            print("Test complete.")

if __name__ == "__main__":
    asyncio.run(main())
